examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImSubscribeCommand.java

```diff
@@ -0,0 +1,113 @@
+package com.matter.controller.commands.pairing;
+
+import chip.devicecontroller.ChipDeviceController;
+import chip.devicecontroller.GetConnectedDeviceCallbackJni.GetConnectedDeviceCallback;
+import chip.devicecontroller.ReportCallback;
+import chip.devicecontroller.SubscriptionEstablishedCallback;
+import chip.devicecontroller.ResubscriptionAttemptCallback;
+import chip.devicecontroller.model.NodeState;
+import chip.devicecontroller.model.ChipAttributePath;
+import chip.devicecontroller.model.ChipEventPath;
+import com.matter.controller.commands.common.CredentialsIssuer;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.ArrayList;
+import javax.annotation.Nullable;
+
+public final class PairOnNetworkLongImSubscribeCommand extends PairingCommand {
+  private static Logger logger = Logger.getLogger(PairOnNetworkLongImSubscribeCommand.class.getName());
```

Comment:
Can be final

Reply:
updated with kotlin, please re-check, thanks